x/twap/types/keys.go

```diff
@@ -79,8 +85,11 @@ func ParseTimeFromHistoricalTimeIndexKey(key []byte) time.Time {
 func ParseTimeFromHistoricalPoolIndexKey(key []byte) (time.Time, error) {
 	keyS := string(key)
 	s := strings.Split(keyS, KeySeparator)
-	if len(s) != 5 || s[0] != historicalTWAPPoolIndexNoSeparator {
-		return time.Time{}, fmt.Errorf("Called ParseTimeFromHistoricalPoolIndexKey on incorrectly formatted key: %v", s)
+	if len(s) != expectedLenOfKeySeparators {
+		return time.Time{}, fmt.Errorf("key incorrect length: expected length (%d), actual length (%d)", expectedLenOfKeySeparators, len(s))
```

Comment:
nit: we can probably define global errors to reuse, including in tests

Reply:
I added an errors.go file and added these errors to it to get the ball rolling! Probably should do a separate PR though to flesh out all errors that should be abstracted to this new errors file

Reply:
I was thinking of something like this (Go errors): https://github.com/osmosis-labs/osmosis/blob/8e63d6250eccffb6e060da5afb83a669fd287254/x/twap/store.go#L15-L29

Based on previous discussions, I don't think there is much value in registered errors, especially having recently discovered that changing the registered error codes can cause state breakage. Let me know what you think

Reply:
Should be good to go now, thanks!

Reply:
I am not a fan of this decision, way more cognitive overhead in code, for unclear reasons to me